addons/delivery_shipper/models/delivery_carrier.py

```diff
@@ -0,0 +1,245 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from markupsafe import Markup
+from odoo.exceptions import UserError
+from .shipper_service import Shipper
+import logging
+import math
+
+logger = logging.getLogger(__name__)
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = 'delivery.carrier'
+
+    delivery_type = fields.Selection(selection_add=[
+        ('shipper', 'Shipper'),
+    ], ondelete={'shipper': lambda records: records.write({'delivery_type': 'fixed', 'fixed_price': 0})})
+
+    shipper_api_key = fields.Char(help='Shipper API Integration key')
```

Comment:
You need to add a neutralization script to replace this key when a database is duplicated.